### PR TITLE
unknown modifications for mzIdentML

### PR DESCRIPTION
**Pull request by hroest**
- allow unknown modifications for mzIdentML 
- bugfixes in mzIdentML code

### CODE REVIEW

**Comment by hroest** [comment]
basically fixed some errors when trying to parse an mzid file produced by crux. Follow up to #2561

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp [hunk 1]
```diff
@@ -225,57 +226,55 @@ namespace OpenMS
...
-        // 0. AnalysisSoftware {1,unbounded}
+        // 0. AnalysisSoftwareList {0,1}
         DOMNodeList* analysisSoftwareElements = xmlDoc->getElementsByTagName(XMLString::transcode("AnalysisSoftware"));
-        if (!analysisSoftwareElements) throw(runtime_error("No AnalysisSoftware nodes"));
         parseAnalysisSoftwareList_(analysisSoftwareElements);
 
         // 1. DataCollection {1,1}
         DOMNodeList* spectraDataElements = xmlDoc->getElementsByTagName(XMLString::transcode("SpectraData"));
-        if (!spectraDataElements) throw(runtime_error("No SpectraData nodes"));
+        // if (spectraDataElements->getLength() == 0) throw(runtime_error("No SpectraData nodes")); // temporarily disabled
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp]: todo?

File: src/openms/source/CHEMISTRY/AASequence.cpp [hunk 2]
```diff
@@ -1121,7 +1123,11 @@ namespace OpenMS
         // create new modification
         ResidueModification * new_mod = new ResidueModification();
         new_mod->setFullId(modification_name); // setting FullId but not Id makes it a user-defined mod
-        new_mod->setOrigin(aas.peptide_.back()->getOneLetterCode()[0]);
+
+        // We cannot set origin if we want to use the same modification name
+        // also at other AA (and since we have no information here, it is safer
+        // to assume that this may happen).
+        // new_mod->setOrigin(aas.peptide_.back()->getOneLetterCode()[0]);
```
Comment (timosachsenberg) [inline src/openms/source/CHEMISTRY/AASequence.cpp]: is this set to X then?

Comment (hroest) [reply src/openms/source/CHEMISTRY/AASequence.cpp]: yes, see https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/CHEMISTRY/ResidueModification.cpp#L50

Comment (hroest) [reply src/openms/source/CHEMISTRY/AASequence.cpp]: so basically we allow this mod on any residue...

**Comment by timosachsenberg** [comment]
Looks good to me

**Comment by lars20070** [comment]
Very nice. Thanks!

**Pull request merged**